### DISCUSSION TITLE
busctl get-property de.pengutronix.rauc

### DISCUSSION THREAD

**Discussion by ghost** [rauc/rauc#1224] [General]
We are wanting to use the busctl get-property de.pengutronix.rauc command to filter on Operation Progress LastError as shown below
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Operation 
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Progress
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer LastError

Please can we have a list all return types from the following
Operation 
Progress
LastError

**Comment by ejoerns** [comment]
Is https://rauc.readthedocs.io/en/latest/reference.html#d-bus-api what you are looking for?

**Reply by ghost** [reply]
not really I want to know what are the return types from the following 
Operation
Progress
LastError

For example if we issue the command busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Operation when not performing a download we get the following response
s "idle"

Please can you provide us a list of all the return types (results) from Operation Progress LastError

**Reply by ejoerns** [reply]
The return **types** (like "s" for string) are fixed and noted in the docs.

Do you mean the possible **values** they can have?

It depends. The possible values for `Operation` are described in the link I provided: Just `idle` or `ìnstalling` at the moment.
The `Progress` of course depends on what you install.
For `LastError` if you refer to e.g. error codes, I have to point you to https://github.com/rauc/rauc/issues/690. Otherwise is is just the text of the last error where we do not have a comprehensive list for.

**Reply by ghost** [reply]
Hi 

Thanks for the update.
I shall go through the link you sent me to get an understanding of the error codes
